ivy/functional/backends/tensorflow/general.py

```diff
@@ -290,6 +291,16 @@ def inplace_update(
         return val
 
 
+def _check_inplace_mode():
```

Comment:
I think this error message can be integrated into line [247](https://github.com/unifyai/ivy/pull/22305/files#diff-08657e30ab249a75e71b4d0cb464494dc19ae11c64604c22d61017f988c809b2R247) by adding an additional condition to the `if` block, same applies to the other backends.
Also, we should create a new Exception class for this like `InplaceUpdateException` in `exceptions.py` and throw that instead.